lib/src/main/scala/spinal/lib/fsm/State.scala

```diff
@@ -219,18 +219,18 @@ class StateMachineSharableRegUInt {
   * }}}
   *
   */
-class StateDelay(cyclesCount: UInt)(implicit stateMachineAccessor: StateMachineAccessor) extends State with StateCompletionTrait {
```

Comment:
This modification seems to make the StateDelay's delay count unchangable.

Reply:
The motivation for this change is that spinal doesn't seem to have something like constant folding, causing the counter reset of the generated hdl to be split into two parts - the reset value comes at the very beginning of the file while the assigement at the end.

Reply:
I've changed the delay count api to be compatible with the previous `Int`, `UInt` or `TimeNumber` inputs.

Reply:
Ahhh interresting ^^ 
Seems ok to me now